File: tvr-core/src/main/java/io/github/binaryfoo/decoders/DecodeSession.java
package io.github.binaryfoo.decoders;

import io.github.binaryfoo.TagMetaData;
import io.github.binaryfoo.decoders.apdu.APDUCommand;
import io.github.binaryfoo.tlv.Tag;

import java.util.HashMap;

public class DecodeSession extends HashMap<Tag, String> {

    private boolean firstGenerateACCommand = true;
    private TagMetaData tagMetaData;
    private APDUCommand command;

    public boolean isFirstGenerateACCommand() {
        return firstGenerateACCommand;
    }

    public void setFirstGenerateACCommand(boolean firstGenerateACCommand) {
        this.firstGenerateACCommand = firstGenerateACCommand;
    }

    public TagMetaData getTagMetaData() {
        return tagMetaData;
    }

    public void setTagMetaData(TagMetaData tagMetaData) {
        this.tagMetaData = tagMetaData;
    }

    public void setCurrentCommand(APDUCommand command) {
        this.command = command;
    }

    public APDUCommand getCommand() {
        return command;
    }
}

File: src/main/java/io/github/binaryfoo/decoders/DecodeSession.kt
package io.github.binaryfoo.decoders

import io.github.binaryfoo.TagMetaData
import io.github.binaryfoo.crypto.RecoveredPublicKeyCertificate
import io.github.binaryfoo.decoders.apdu.APDUCommand
import io.github.binaryfoo.tlv.CommonVendorErrorMode
import io.github.binaryfoo.tlv.Tag
import io.github.binaryfoo.tlv.TagRecognitionMode
import java.util.*

class DecodeSession : HashMap<Tag, String>() {

    var firstGenerateACCommand: Boolean = true
    var tagMetaData: TagMetaData? = null
    var currentCommand: APDUCommand? = null
    var issuerPublicKeyCertificate: RecoveredPublicKeyCertificate? = null
    var iccPublicKeyCertificate: RecoveredPublicKeyCertificate? = null
    var signedDynamicAppData: String? = null
    var tagRecognitionMode: TagRecognitionMode = CommonVendorErrorMode

}
